pkg/routes/logs_test.go

```diff
@@ -0,0 +1,57 @@
+package routes
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+)
+
+func TestPreCheckLogFileNameLength(t *testing.T) {
+	oversizeFileName := fmt.Sprintf("%0256s", "a")
+	correctFileName := fmt.Sprintf("%0255s", "a")
+
+	// check oversize filename
+	isOversize := preCheckLogFileNameLength(oversizeFileName)
+	if !isOversize {
+		t.Error("failed to check oversize filename")
+	}
+
+	// check correct filename when file is not exist
+	NotOversize := preCheckLogFileNameLength(correctFileName)
+	if NotOversize {
+		t.Error("failed to check allowable length filename")
+	}
+
+	// create test file with correct name and check it
+	_, err := os.Create(correctFileName)
+	if err != nil {
+		t.Error("failed to create test file")
+	}
+	defer os.Remove(correctFileName)
+	NotOversize = preCheckLogFileNameLength(correctFileName)
+	if NotOversize {
+		t.Error("failed to check allowable length filename")
+	}
+}
+
+func TestHttpServeFile(t *testing.T) {
+	logFilePath := "/var/log"
+	oversizeFileName := fmt.Sprintf("%s/%0256s", logFilePath, "a")
+	correctFileName := fmt.Sprintf("%s/%0255s", logFilePath, "a")
+
+	request, _ := http.NewRequest("", "", nil)
+
+	w1 := httptest.NewRecorder()
+	http.ServeFile(w1, request, oversizeFileName)
+	if w1.Code != http.StatusInternalServerError {
+		t.Errorf("expected response code 500, get %d", w1.Code)
+	}
+
+	w2 := httptest.NewRecorder()
+	http.ServeFile(w2, request, correctFileName)
+	if w2.Code != http.StatusNotFound {
```

Comment:
for correctFileName, the response code should be 200, no?

Reply:
Sure, fixed.